requestmanager/requestmanager.go

```diff
@@ -398,61 +421,24 @@ func (rm *RequestManager) processBlockHooks(p peer.ID, response graphsync.Respon
 	return result.Err
 }
 
-func (rm *RequestManager) setupRequest(p peer.ID, request gsmsg.GraphSyncRequest, hooksResult hooks.RequestResult, selector selector.Selector) (chan graphsync.ResponseProgress, chan error) {
-	networkErrorChan := make(chan error, 1)
-	ctx, cancel := context.WithCancel(rm.ctx)
-	requestStatus := &inProgressRequestStatus{
-		ctx: ctx, cancelFn: cancel, p: p, networkError: networkErrorChan,
-	}
-	lastResponse := &requestStatus.lastResponse
-	rm.inProgressRequestStatuses[request.ID()] = requestStatus
-	lastResponse.Store(gsmsg.NewResponse(request.ID(), graphsync.RequestAcknowledged))
-	rm.peerHandler.SendRequest(p, request)
-	inProgressChan := make(chan graphsync.ResponseProgress)
-	inProgressErr := make(chan error)
-	go func() {
-		loaderFn := loader.WrapAsyncLoader(ctx, rm.asyncLoader.AsyncLoad, request.ID(), inProgressErr, func(bd graphsync.BlockData) error {
-			response := lastResponse.Load().(gsmsg.GraphSyncResponse)
-			return rm.processBlockHooks(p, response, bd)
-		})
-		visitor := visitToChannel(ctx, inProgressChan)
-		_ = ipldutil.Traverse(ctx, loaderFn, hooksResult.CustomChooser, cidlink.Link{Cid: request.Root()}, selector, visitor)
-		select {
-		case networkError := <-networkErrorChan:
-			select {
-			case <-rm.ctx.Done():
-			case inProgressErr <- networkError:
-			}
-		default:
-		}
-		select {
-		case <-ctx.Done():
-		case rm.messages <- &terminateRequestMessage{request.ID()}:
-		}
-		close(inProgressChan)
-		close(inProgressErr)
-	}()
-	return inProgressChan, inProgressErr
-}
-
-func (rm *RequestManager) validateRequest(requestID graphsync.RequestID, p peer.ID, root ipld.Link, selectorSpec ipld.Node, extensions []graphsync.ExtensionData) (gsmsg.GraphSyncRequest, hooks.RequestResult, selector.Selector, error) {
+func (rm *RequestManager) validateRequest(requestID graphsync.RequestID, p peer.ID, root ipld.Link, selectorSpec ipld.Node, extensions []graphsync.ExtensionData) (gsmsg.GraphSyncRequest, hooks.RequestResult, error) {
 	_, err := ipldutil.EncodeNode(selectorSpec)
 	if err != nil {
-		return gsmsg.GraphSyncRequest{}, hooks.RequestResult{}, nil, err
+		return gsmsg.GraphSyncRequest{}, hooks.RequestResult{}, err
 	}
-	selector, err := ipldutil.ParseSelector(selectorSpec)
+	_, err = ipldutil.ParseSelector(selectorSpec)
 	if err != nil {
-		return gsmsg.GraphSyncRequest{}, hooks.RequestResult{}, nil, err
+		return gsmsg.GraphSyncRequest{}, hooks.RequestResult{}, err
```

Comment:
Looks great! my only nit would be that I'd rewire this error check into an inlined version. Otherwise it seems like `.ParseSelector` has side-effects ( also do you even need to check `.EncodeNode` above? )

Reply:
oh shoot. just realized what you're saying. will fix in later PR